### PR TITLE
system variable: add tidb_allow_fallback_to_tikv

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add `tidb_allow_fallback_to_tikv` into system variables.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: pingcap/docs-cn#5484
- Other reference link(s): pingcap/tidb#22459, pingcap/tidb#22886

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1074,3 +1074,9 @@ This variable is an alias for _transaction_isolation_.
 - Scope: SESSION | GLOBAL
 - Default value: ON
 - This variable controls whether to use the high precision mode when computing the window functions.
+
+### tidb_enable_engine_fallback
+
+- Scope: SESSION | GLOBAL
+- Default value: OFF
+- This variable controls whether to retry execution with another engine when the used engine's failure causes execution failure of SQL statement. If the variable turns on, currently TiDB supports retrying execution with TiKV when TiFlash's failure causes execution failure of SQL statement.
```
Comment (morgo) [inline system-variables.md:1082]: The options in this page are in alphabetical order (the order which they were added to the server is not useful to end-users). Please move it to the correct place :-)

File: system-variables.md [hunk 2]
```diff
@@ -172,6 +172,12 @@ mysql> SELECT * FROM t1;
     * `1`: Aggregation and join requests are sent in batches
     * `2`: All coprocessor requests are sent in batches
 
+### tidb_allow_fallback_to_tikv <span class="version-mark">New in v5.0 GA version</span>
+
+- Scope: SESSION | GLOBAL
+- Default value: ""
+- When the execution of an SQL statement fails due to a failure of the storage engine specified by this variable, TiDB will use TiKV to execute the SQL statement again. Currently, this variable can be set to "" or "tiflash". When this variable is set to "tiflash", TiDB will retry with TiKV if the execution of the SQL statement fails due to a failure of TiFlash.
```
Comment (qiancai) [inline system-variables.md:179]:
```suggestion
- This variable is used to specify a storage engine. If the execution of an SQL statement fails due to a failure of the specified storage engine, TiDB will use TiKV to execute the SQL statement again. This variable can be set to "" or "tiflash". When this variable is set to "tiflash", if the execution of an SQL statement fails due to a failure of TiFlash, TiDB will retry executing the SQL statement with TiKV.
```

File: system-variables.md [hunk 3]
```diff
@@ -172,6 +172,12 @@ mysql> SELECT * FROM t1;
     * `1`: Aggregation and join requests are sent in batches
     * `2`: All coprocessor requests are sent in batches
 
+### tidb_allow_fallback_to_tikv <span class="version-mark">New in v5.0 GA version</span>
```
Comment (TomShawn) [inline system-variables.md:175]:
```suggestion
### tidb_allow_fallback_to_tikv <span class="version-mark">New in v5.0 GA</span>
```

File: system-variables.md [hunk 4]
```diff
@@ -172,6 +172,12 @@ mysql> SELECT * FROM t1;
...
+
+- Scope: SESSION | GLOBAL
+- Default value: ""
+- This variable is used to specify a storage engine. If the execution of an SQL statement fails due to a failure of the specified storage engine, TiDB use TiKV to execute the SQL statement again. This variable can be set to "" or "tiflash". When this variable is set to "tiflash", if the execution of an SQL statement fails due to a failure of TiFlash, TiDB retry executing the SQL statement with TiKV.
```
Comment (TomShawn) [inline system-variables.md:179]:
```suggestion
- This variable is used to specify a storage engine. If the execution of an SQL statement fails due to a failure of the specified storage engine, TiDB retries executing this SQL statement with TiKV. This variable can be set to "" or "tiflash". When this variable is set to "tiflash", if the execution of an SQL statement fails due to a failure of TiFlash, TiDB retries executing the SQL statement with TiKV.
```

File: system-variables.md [hunk 5]
```diff
@@ -172,6 +172,12 @@ mysql> SELECT * FROM t1;
     * `1`: Aggregation and join requests are sent in batches
     * `2`: All coprocessor requests are sent in batches
 
+### tidb_allow_fallback_to_tikv <span class="version-mark">New in v5.0 GA</span>
+
+- Scope: SESSION | GLOBAL
+- Default value: ""
+- This variable is used to specify a storage engine. If the execution of an SQL statement fails due to a failure of the specified storage engine, TiDB retries executing this SQL statement with TiKV. This variable can be set to "" or "tiflash". When this variable is set to "tiflash", if the execution of an SQL statement fails due to a failure of TiFlash, TiDB retries executing the SQL statement with TiKV.
```
Comment (TomShawn) [inline system-variables.md:179]:
```suggestion
- This variable is used to specify a storage engine list that uses TiKV as the backup storage engine. If the execution of a SQL statement fails due to a failure of the specified storage engine in the list, TiDB retries executing this SQL statement with TiKV. This variable can be set to "" or "tiflash". When this variable is set to "tiflash", if the execution of a SQL statement fails due to a failure of TiFlash, TiDB retries executing this SQL statement with TiKV.
```

**Comment by yikeke** [comment]
@morgo PTAL again

**Pull request merged**